packages/blocks/src/api/validation/index.js

```diff
@@ -711,28 +871,59 @@ export function isEquivalentHTML( actual, expected, logger = createLogger() ) {
  */
 
 /**
- * Returns an object with `isValid` property set to `true` if the parsed block
- * is valid given the input content. A block is considered valid if, when serialized
- * with assumed attributes, the content matches the original value. If block is
- * invalid, this function returns all validations issues as well.
+ * Returns validation results for a parsed block. The validation system uses a
+ * 6-level classification to determine content integrity with varying degrees of
+ * confidence, from perfect match (ValidBlock) to requiring user intervention (InvalidBlock).
  *
- * @param {WPBlock}            block                          block object.
+ * @param {WPBlock}            block                          Block object.
  * @param {WPBlockType|string} [blockTypeOrName = block.name] Block type or name, inferred from block if not given.
  *
- * @return {[boolean,Array<LoggerItem>]} validation results.
+ * @return {[boolean, Array<LoggerItem>, Object]} Validation results tuple:
+ *                                                - isValid: boolean indicating if block is valid
+ *                                                - issues: array of validation issues
+ *                                                - metadata: object containing validationLevel, originalContent, generatedContent
  */
 export function validateBlock( block, blockTypeOrName = block.name ) {
 	const isFallbackBlock =
 		block.name === getFreeformContentHandlerName() ||
 		block.name === getUnregisteredTypeHandlerName();
 
-	// Shortcut to avoid costly validation.
+	// Shortcut: Fallback blocks (freeform/unregistered) are marked as raw transformed.
 	if ( isFallbackBlock ) {
```

Comment:
I wonder if this should apply to both unregistered and freeform? Feels like it should apply only to "freeform". Also I feel like it should apply only when there's no comment delimiters for the block, in other words, if the block comment explicitly states that it should use core/classic or core/html for instance, it shouldn't auto-transform even if that block is marked as "freeform" block content handler.

Reply:
I revised this logic a bit. It comes down to the fact we wrap unregistered blocks in `core/missing` which handles the UI. A block with `core/missing` is "valid" from the perspective of the validation flow.